### PR TITLE
Add inference example for bundle

### PR DESCRIPTION
**Pull request by Nic-Ma**
part of #486 

### Description
This PR added inference example for monai.bundle.

### Status
**Ready**

### Checks

- [ ] Notebook runs automatically `./runner [-p <regex_pattern>]`

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Command to run from shell:
```
python -m monai.bundle.scripts run --meta_file="configs/metadata.json" --config_file="configs/inference.json" --target="evaluator"
```

File: modules/bundles/spleen_segmentation/configs/inference.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                "b_max": 1,
+                "clip": true
+            },
+            {
+                "_target_": "EnsureTyped",
+                "keys": "image"
+            }
+        ]
+    },
+    "datalist": {
```
Comment (wyli) [inline modules/bundles/spleen_segmentation/configs/inference.json:52]: To help onboard the pytorch users to get started, I believe the example should follow the basic workflow such as https://github.com/Project-MONAI/tutorials/tree/master/3d_segmentation/ignite . So I think we should simplify and remove this datalist section and other utilities that are task/dataset specific

Comment (Nic-Ma) [reply modules/bundles/spleen_segmentation/configs/inference.json]: Updated in the latest commit.

Thanks.

**Comment by Nic-Ma** [comment]
PR is ready for review, it depends on: https://github.com/Project-MONAI/MONAI/pull/3994.

Thanks.

File: modules/bundles/spleen_segmentation/configs/metadata.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    "monai_version": "0.8.0",
+    "pytorch_version": "1.10.0",
+    "numpy_version": "1.21.2",
+    "optional_packages_version": {
+        "nibabel": "3.2.1"
+    },
+    "task": "Decathlon spleen segmentation",
+    "description": "A pre-trained model for volumetric (3D) segmentation of the spleen from CT image",
+    "authors": "MONAI team",
```
Comment (wyli) [inline modules/bundles/spleen_segmentation/configs/metadata.json:16]: how to support this metadata? https://citation-file-format.github.io/about/

Comment (Nic-Ma) [reply modules/bundles/spleen_segmentation/configs/metadata.json:16]: CC @ericspod for the metadata format discussion.

Thanks.

Comment (ericspod) [reply modules/bundles/spleen_segmentation/configs/metadata.json:16]: `CITATION.cff` would be another file in the bundle that can be optionally added and referred to a `citation` value in the metadata?

**Review by wyli — APPROVED** [review]

**Pull request merged**